cheatsheets_draft/Webhook_Security_Guidelines_CheatSheet.md

```diff
@@ -0,0 +1,169 @@
+# Webhook Security Guidelines Cheat Sheet
+
+## Introduction
+
+Webhooks provide a mechanism where a Server-side application can notify a client-side application when a new even (that the client-side application might be interested in) has occurred on the server. Webhooks are incredibly useful and a resource-light way to implement event reactions. However, webhooks can also be abused and the following considerations must be taken into account to secure Webhooks. It should also be noted that webhook security considerations apply to both publisher/server and subscriber/client. 
+
+At the core of webhooks fundamental security best practices are available however are not enabled by default. This is a gap in the “default secure” principle however with focus and configuration this gap can be addressed. One item to note is webhook urls by default carry a high level of confidentiality. This comes from the lack of default secure principle mentioned above. With this webhook urls should be kept secret and while the security community has generally accepted that obscurity is not security it is a best practice to follow. If your webhooks are sharing sensitive information the following considerations should be reviewed and implemented.
+
+At high level, the following requirements outline the security mechanisms in various ways. These high level requirements allow engineers and developers to properly implement security controls. 
+
+
+- **Confidentiality: A passive attacker must not be able to read Webhook Notifications, as they may contain confidential data.**
```

Comment:
Thank you @ThunderSon, I agree the guidelines should help any developer building Webhooks rather than Attacker developing abuse cases. I will remove CIA and make it high level requirement as:

- Webhook URLs should treated like any other secrets like password, Keys, API token and should not be hardcoded in source code. 
- It's important to secure Webhook URLs in Vault than storing it in Version Control System.
- Communication between Client and Webhook server should be mutually Authenticated - TLS Authentication
- Webhook requests should be signed and signature should be validated
- Follow Payload best practices and prevent replay attacks.